catalog/src/main/java/it/eng/catalog/service/ArtifactService.java

```diff
@@ -37,18 +34,51 @@ public class ArtifactService {
 	
 	private final MongoTemplate mongoTemplate;
 	private final DatasetService datasetService;
+	private final ArtifactRepository artifactRepository;
 	
-	public ArtifactService(MongoTemplate mongoTemplate, DatasetService datasetService) {
+	public List<Artifact> getArtifacts(String artifactId) {
+		if(StringUtils.isNotBlank(artifactId)) {
+			return artifactRepository.findById(artifactId)
+					.stream()
+					.collect(Collectors.toList());
+		}
+		return artifactRepository.findAll();
+	}
+	
+	public ArtifactService(MongoTemplate mongoTemplate, DatasetService datasetService, ArtifactRepository artifactRepository) {
```

Comment:
Agreed regarding constructor. As for the naming, a bigger refactoring has to be done since all service classes in catalog have API logic, except from Catalog and Dataset services which also have protocol logic and have to be split. How do you want me to proceed with this?

Reply:
We will postpone renaming, since other services are not named API.